backend/lib/endpoints/users.js

```diff
@@ -12,34 +12,49 @@ const {
 async function routes(app) {
   const User = app.mongo.model("IndividualUser");
 
-  app.get("/current", { preValidation: [app.authenticate] }, async (req) => {
-    const result = await User.findById(req.userId);
-    if (result === null) {
-      throw app.httpErrors.notFound();
+  app.get(
+    "/current",
+    { preValidation: [app.authenticateOptional] },
+    async (req) => {
+      const { userId } = req;
+
+      // don't error if not authenticated
+      // client always queries on start to check if authenticated
+      if (!userId) return null;
```

Comment:
not sure if this is a good reason not to return an error? I think it might be better to keep this still as a `401` and make sure that the client handles that response appropriately

Reply:
Yeah I wasn't sure on this one - my concern was having an errorred returned every time the app starts up. But actually I think what would be better now that we have 2nd cookie - only query /users/current/ on start-up if there is the non-httpOnly remember token present.

Reply:
@robinv85 I fixed the 3 other minor issues and implemented an alternate approach for this issue. Take a look and let me know if this is good (it seems to work nicely). The final suggestion  I think the current state is good but open to other suggestions.